src/Auth0.ManagementApi/HttpClientManagementConnection.cs

```diff
@@ -166,5 +179,52 @@ private static string SerializeFormBodyValue(object value)
             if (value is bool boolean) return boolean ? "true" : "false";
             return Uri.EscapeDataString(value.ToString());
         }
+
+        private async Task<TResult> Retry<TResult>(Func<Task<TResult>> retryable)
+        {
+            int? configuredNrOfTries = options.NumberOfHttpRetries;
+            var DEFAULT_NUMBER_RETRIES = 3;
+            var MAX_NUMBER_RETRIES = 10;
+            var BASE_DELAY = 100;
+            var nrOfTries = 0;
+
+            var nrOfTriesToAttempt = Math.Min(MAX_NUMBER_RETRIES, configuredNrOfTries ?? DEFAULT_NUMBER_RETRIES);
+
+            while (true)
+            {
+                try
+                {
+                    nrOfTries++;
+
+                    return await retryable();
+                }
+                catch (Exception ex)
+                {
+                    if (!(ex is RateLimitApiException) || nrOfTries >= nrOfTriesToAttempt)
+                    {
+                        throw;
+                    }
+                }
+
+                // Use an exponential back-off with the formula:
+                // max(MIN_REQUEST_RETRY_DELAY, min(MAX_REQUEST_RETRY_DELAY, (BASE_DELAY * (2 ** attempt - 1)) + random_between(0, MAX_REQUEST_RETRY_JITTER)))
+                //
+                // ✔ Each attempt increases base delay by (100ms * (2 ** attempt - 1))
+                // ✔ Randomizes jitter, adding up to MAX_REQUEST_RETRY_JITTER (250ms)
+                // ✔ Never less than MIN_REQUEST_RETRY_DELAY (100ms)
+                // ✔ Never more than MAX_REQUEST_RETRY_DELAY (500ms)
+                //
+                var MAX_REQUEST_RETRY_JITTER = 250;
+                var MAX_REQUEST_RETRY_DELAY = 500;
+                var MIN_REQUEST_RETRY_DELAY = 100;
+
+                var wait = Convert.ToInt32(BASE_DELAY * Math.Pow(2, nrOfTries - 1));
+                wait = new Random().Next(wait + 1, wait + MAX_REQUEST_RETRY_JITTER);
```

Comment:
This might be a bit of an overhead in high-throughput .NET Framework applications, per:
https://docs.microsoft.com/en-us/dotnet/api/system.random?view=net-5.0#avoiding-multiple-instantiations

Polly implemented a [ConcurrentRandom ](https://github.com/Polly-Contrib/Polly.Contrib.WaitAndRetry/blob/master/src/Polly.Contrib.WaitAndRetry/ConcurrentRandom.cs)class to deal with this issue.

Reply:
That makes sense. Would have loved to use polly, but couldn't find a non-breaking way to bring in polly here. That said, I am not sure the thread safety is an issue here, as that would mean we have two (or more) random generators, meaning we can be generating the same values ... Which is fine. We are not looking for unique random numbers here.

Therefore I believe moving the random instance to class level here could do the trick to increase performance impact.

Reply:
Random returns `0` if not used in a thread safe way, not the same number, which probably isn't ideal in this scenario. 

btw the constants here could be lifted to the class level as well.

I'm guessing there's no near-term plan for a v8 that'd allow for some breaking changes?

Reply:
Updated version looks great 👍 

This issue will thankfully be dealt with out of the box in .NET 6 with the introduction of `Random.Shared`.